From df393819b9a040b183e1c7e8b9e42a97cb711f55 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Mon, 15 Jul 2019 16:24:35 -0400
Subject: [PATCH] gpg: drop import-clean from default keyserver import options

* g10/gpg.c (main): drop IMPORT_CLEAN from the
default opt.keyserver_options.import_options
* doc/gpg.texi: reflect this change in the documentation

Given that SELF_SIGS_ONLY is already set, it's not clear what
additional benefit IMPORT_CLEAN provides.  Furthermore, IMPORT_CLEAN
means that receiving an OpenPGP certificate from a keyserver will
potentially delete data that is otherwise held in the local keyring,
which is surprising to users who expect retrieval from the keyservers
to be purely additive.

GnuPG-Bug-Id: 4628
Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
---
 doc/gpg.texi | 2 +-
 g10/gpg.c    | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/doc/gpg.texi b/doc/gpg.texi
index 7e80a293a..652003c27 100644
--- a/doc/gpg.texi
+++ b/doc/gpg.texi
@@ -2188,7 +2188,7 @@ are available for all keyserver types, some common options are:
 
 @end table
 
-The default list of options is: "self-sigs-only, import-clean,
+The default list of options is: "self-sigs-only,
 repair-keys, repair-pks-subkey-bug, export-attributes". However, if
 the actual used source is an LDAP server "no-self-sigs-only" is
 assumed unless "self-sigs-only" has been explicitly configured.
diff --git a/g10/gpg.c b/g10/gpg.c
index 28f47f80b..03298a295 100644
--- a/g10/gpg.c
+++ b/g10/gpg.c
@@ -2485,8 +2485,7 @@ main (int argc, char **argv)
 					    | IMPORT_REPAIR_PKS_SUBKEY_BUG
                                             | IMPORT_SELF_SIGS_ONLY
                                             | IMPORT_COLLAPSE_UIDS
-                                            | IMPORT_COLLAPSE_SUBKEYS
-                                            | IMPORT_CLEAN);
+                                            | IMPORT_COLLAPSE_SUBKEYS);
     opt.keyserver_options.export_options = EXPORT_ATTRIBUTES;
     opt.keyserver_options.options = 0;
     opt.verify_options = (LIST_SHOW_UID_VALIDITY
